### PR TITLE
Randomize Projects List

### LINKED ISSUE #185: Randomize Projects List
The projects list on the projects can be randomized each time for better discovery.

### PR DESCRIPTION
**Pull request by asmitdeb**
Closes #185
Added a function which randomly shuffles the projects array

### CODE REVIEW

File: src/pages/Projects.tsx [hunk 1]
```diff
@@ -7,6 +7,16 @@ import Fuse from "fuse.js";
 import SpinnerLoader from "../components/SpinnerLoader";
 import { IconContext } from "react-icons";
 
+// function to randomly shuffle an array
```
Comment (harshkhandeparkar) [inline src/pages/Projects.tsx]: Please put this function in a file in `utils/`

File: src/pages/Projects.tsx [hunk 2]
```diff
@@ -30,8 +40,9 @@ function Projects() {
 
   useEffect(() => {
     makeRequest("project", "get")
-      .then((response) => {
+      .then(async (response) => {
```
Comment (harshkhandeparkar) [inline src/pages/Projects.tsx]: I don't think async is required here.

File: src/pages/Projects.tsx [hunk 3]
```diff
@@ -30,8 +40,9 @@ function Projects() {
...
         if (response.is_ok) {
+          await shuffle(response.response);// shuffles the array
```
Comment (harshkhandeparkar) [inline src/pages/Projects.tsx]: Here as well, await won't make a difference.

**Review by harshkhandeparkar — CHANGES_REQUESTED** [review]

File: src/pages/Projects.tsx (hunk 1 above)
Comment (asmitdeb) [reply src/pages/Projects.tsx]: Done

**Comment by asmitdeb** [comment]
Please review this commit @harshkhandeparkar

File: src/pages/Projects.tsx [hunk 4]
```diff
@@ -3,10 +3,12 @@ import { MdCancel } from "react-icons/md";
 import ProjectCard from "../components/ProjectCard";
 import { makeRequest } from "../util/backend";
 import { IEndpointTypes } from "../util/types";
+import { shuffle } from "../util/shuffle";
 import Fuse from "fuse.js";
 import SpinnerLoader from "../components/SpinnerLoader";
 import { IconContext } from "react-icons";
 
+
```
Comment (harshkhandeparkar) [inline src/pages/Projects.tsx]: Please remove this extra space.

File: src/util/shuffle.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
+// function to randomly shuffle an array
+export function shuffle(array) {
```
Comment (harshkhandeparkar) [inline src/util/shuffle.ts]: It would be better if this function creates a new array and returns it rather than shuffling the array in place.

**Review by harshkhandeparkar — CHANGES_REQUESTED** [review]
Please run `pnpm build` locally and fix all the errors it returns.

File: src/pages/Projects.tsx (hunk 4 above)
Comment (asmitdeb) [reply src/pages/Projects.tsx]: Done

File: src/util/shuffle.ts (hunk 5 above)
Comment (asmitdeb) [reply src/util/shuffle.ts]: Done, please check @harshkhandeparkar

File: src/util/shuffle.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
+// function to randomly shuffle an array
+export function shuffle(array: any[]):any[] {
+    const arraycopy: any[]= [];
+    array.forEach((val: any) => arraycopy.push(Object.assign({}, val)));
+    for (var i = arraycopy.length - 1; i > 0; i--) {
+        var j = Math.floor(Math.random() * (i + 1));
```
Comment (harshkhandeparkar) [inline src/util/shuffle.ts]: It would be better to use let and const here.

**Review by harshkhandeparkar — CHANGES_REQUESTED** [review]

File: src/util/shuffle.ts (hunk 6 above)
Comment (asmitdeb) [reply src/util/shuffle.ts]: You mean `let arraycopy`  and `const temp` right? Or do you want me to replace `var` with `let`?
@harshkhandeparkar

Comment (harshkhandeparkar) [reply src/util/shuffle.ts]: Use let only if the variable is reassigned, const otherwise.

Comment (asmitdeb) [reply src/util/shuffle.ts]: Done, @harshkhandeparkar

**Review by harshkhandeparkar — APPROVED** [review]

**Comment by harshkhandeparkar** [comment]
Thank you @asmitdeb! :rocket:

**Pull request merged**